trtools/utils/tests/test_trharmonizer.py

```diff
@@ -289,183 +378,199 @@ def test_GetGenotypeCounts():
+        al_counts_idx_slist.items()
+    ) and len(al_counts_idx_slist) == len(true_idx_al_counts_slist))
+    assert (all(
+        v == true_len_al_counts_slist[k] for k, v in
+        al_counts_uselength_slist.items()
     ) and len(al_counts_uselength_slist) == len(true_len_al_counts_slist))
     assert (all(
         v == true_al_counts_slist[k] for k, v
         in al_counts_nolength_slist.items()
     ) and len(al_counts_nolength_slist) == len(true_al_counts_slist))
 
-    # Test example where alt=[]
-    rec = trh.TRRecord(dummy_record3, ref_allele, [], "CAG", "", None)
-    true_len_al_counts = {3: 9}
-    al_counts_uselength = rec.GetAlleleCounts()
-    assert (all(v == true_len_al_counts[k] for k,
-                v in al_counts_uselength.items()) and len(al_counts_uselength) == len(true_len_al_counts))
-
-    # Test example with non of samples in samplelist in VCF
-    rec = trh.TRRecord(dummy_record3, ref_allele, [], "CAG", "", None)
-    true_len_al_counts_slist = {}
-    al_counts_uselength_slist = rec.GetAlleleCounts(samplelist = ['NonExistentSample'])
-    assert (all(v == true_len_al_counts_slist[k] for k,v in al_counts_uselength_slist.items()) and len(al_counts_uselength_slist) == len(true_len_al_counts_slist))
-
-    # Test example where that has one uncalled sample only
-    rec = trh.TRRecord(dummy_record4, ref_allele, [], "CAG", "", None)
-    true_len_al_counts = {}
-    al_counts_uselength = rec.GetAlleleCounts()
-    assert (all(v == true_len_al_counts[k] for k,v in al_counts_uselength.items()) and len(al_counts_uselength) == len(true_len_al_counts))
-
 
 def test_GetAlleleFreqs():
-    # Test good example, no samplelist, uselength=True (default)
-    ref_allele = "CAGCAGCAG"
-    alt_alleles = ["CAGCAGCAGCAG","CAGCAGCAGCAGCAGCAG"]
-    rec = trh.TRRecord(dummy_record1, ref_allele, alt_alleles, "CAG", "", None)
-    print(rec) # To test str function
-    true_al_freqs = {ref_allele: 0.18181818181818182, alt_alleles[0]: 0.5454545454545454, alt_alleles[1]: 0.2727272727272727}
+    # Test working example, no samplelist
+    dummy_record = get_dummy_record()
+    ref_allele = dummy_record.REF
+    alt_alleles = dummy_record.ALT
+    rec = trh.TRRecord(dummy_record, ref_allele, alt_alleles, "CAG", "", None)
+    true_al_freqs = {
+        ref_allele: 0.18181818181818182,
+        alt_alleles[0]: 0.5454545454545454,
+        alt_alleles[1]: 0.2727272727272727
+    }
     true_len_al_freqs = {3: 0.18181818181818182, 4: 0.5454545454545454, 6: 0.2727272727272727}
-    al_freqs_uselength = rec.GetAlleleFreqs()
-    al_freqs_nolength = rec.GetAlleleFreqs(uselength = False)
-    assert (all(v == true_len_al_freqs[k] for k,v in al_freqs_uselength.items()) and len(al_freqs_uselength) == len(true_len_al_freqs))
-    assert (all(v == true_al_freqs[k] for k,v in al_freqs_nolength.items()) and len(al_freqs_nolength) == len(true_al_freqs))
+    true_idx_al_freqs = {0: 0.18181818181818182, 1: 0.5454545454545454, 2: 0.2727272727272727}
```

Comment:
can we round these to 3-4 decimal points when we assert? I'm worried environment changes may cause mismatch. Feel free to ignore if that's not reasonable.

Reply:
Good idea. Pytest's [approx](https://docs.pytest.org/en/latest/reference.html#pytest-approx) seems like a more robust solution to this problem, so I'm using that, and I've truncated the values to confirm that works.